js/ui/handler/box_zoom.js

```diff
@@ -67,7 +67,7 @@ BoxZoom.prototype = {
         this._finish();
 
         this._map
-            .fitBounds(bounds, {linear: true})
+            .fitBounds(bounds, {linear: true, bearing: this._map.getBearing()})
```

Comment:
These bounds are incorrect: if you rotate the map by 45° clockwise and box-zoom the District of Columbia, `bounds.ne` is the northern corner of D.C. and `bounds.sw` is the southern corner. But `Camera.prototype.fitBounds()` is expecting an unrotated bounding box. So the camera ends up trying to fit to a bogus bounding box.

There are two solutions: the simpler solution is to extend `bounds` by the other two corners of the zoom box. The center point will be correct, but the zoom level would be suboptimal in degenerate cases like D.C. The more complex but more correct solution is to port over the changes in mapbox/mapbox-gl-native#1795, then pass in the four corners of the zoom box rather than a `LatLngBounds`.